games/static/games/proto.js

```diff
@@ -0,0 +1,237 @@
+        square.setCharacter(this)
+    }
+}
+
+class Rocket {
+    constructor(imageCssClass, indexOnBoard) {
+        this.imageCssClass = imageCssClass
+        this.indexOnBoard = indexOnBoard
+    }
+    getCssClass() {
+        return this.imageCssClass
+    }
+    moveTo(square) {
+        square.setCharacter(this)
+    }
+}
+
+class Alien {
+    constructor(name, imageCssClass, health, square, lineWidth) {
+        this.name = name
+        this.imageCssClass = imageCssClass
+        this.health = health
+        this.colorByHealthLevel = {
+            1: 'green',
+            2: 'blue'
+        }
+        this.square = square
+        this.lineWidth = lineWidth
+        this.type = 'alien'
+    }
+    getCssClass() {
+        return this.imageCssClass
+    }
+    moveTo(square) {
+        square.setCharacter(this)
+    }
+}
+
+function createInvader(indexOnBoard, lineWidth) {
+    let alien = new Alien('invader', 'invader', 1, indexOnBoard, lineWidth)
+    return alien
+}
+
+function createIntruder(indexOnBoard, lineWidth) {
+    return new Alien('intruder', 'intruder', 2, indexOnBoard, lineWidth)
+}
+
+class Board {
+    constructor(parent, width, setup) {
+        this.parent = parent
+        this.width = width
+        this.aliens = []
+        this.squares = []
+        this.setup = setup
+        this.alienLocations = setup.getAlienLocations()
+        this.direction = 1
+
+    }
+    createChildSquareAddToSquares(index) {
+        let div = document.createElement('div')
+        this.parent.appendChild(div)
+        const borders = []
+        if (index >= this.bottomRow()) {
+            borders.push('ground')
+        } else if (index % this.width === 0) {
+            borders.push('left')
+        } else if (index % this.width === this.width-1) {
+            borders.push('right')
+        }
+        this.squares.push(new Square(index, div, this.width, borders))
+    }
+    bottomRow() {
+        return (this.width * this.width - this.width)
+    }
+    defenderLocation() {
+        return this.width * this.width - (this.width + Math.floor(this.width / 2) + 1)
+    }
+
+    draw() {
+        for (let i = 0; i < this.width * this.width; i++) {
+            this.createChildSquareAddToSquares(i)
+            if (this.alienLocations.includes(i)) {
+                let invader = createInvader(this.squares[i].index, this.width)
+                this.squares[i].setCharacter(invader)
+                // this.aliens.push(invader)
+            }
+            if (i >= this.bottomRow()) {
+                this.squares[i].setGround()
+            }
+        }
+        let defender = new Defender('defender', this.width)
+        this.squares[this.defenderLocation()].setCharacter(defender)
+    }
+
+    
+    move() {
+        const aliensAtLeftBorder = this.squares[this.alienLocations[0]].isLeftBorder()
+        const aliensAtRightBorder = this.squares[this.alienLocations[this.alienLocations.length - 1]].isRightBorder()
```

Comment:
Glancing at the move function i can tell its doing too many things.
First of all, there are too many levels of abstractions. Try to abstract away the details of calculations into functions or objects.
For example calculating the direction can be its own function.
Getting an array of the new locations can be another, and so on.